datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -990,24 +989,44 @@ pub async fn from_substrait_rel(
 fn ensure_schema_compatability(
     table: DataFrame,
     substrait_schema: DFSchema,
-) -> Result<DataFrame> {
-    let df_schema = table.schema().to_owned().strip_qualifiers();
+) -> Result<LogicalPlan> {
+    let df_schema = table.schema().to_owned();
```

Comment:
Ah, the substrait_schema was produced with 
```
from_substrait_named_struct(named_struct, extensions)?
    .replace_qualifier(table_reference.clone());
 ```
so it includes the qualifiers, and that's why this wasn't matching. 

I have a small preference to **not add** the qualifiers to the substrait schema, and then **do** strip them here (given this is a simple read, the qualifier itself _should_ always match unless we've screwed something up, so given we need to either add it to one side or remove from another, I think it'd make more sense to do the comparison **without** it rather than with it. But doesn't matter much either way.

Reply:
yeah, that makes more sense. let me give it a try

Reply:
I'm bumping into some round-trip errors when I try this. I suspect it should be related to optimization calls used in the tests themselves. I'll leave it as is for now and try to fix when I get rid of optimization calls in the tests as well.